pkg/api/customization/catalog/template.go

```diff
@@ -124,3 +152,35 @@ func (t TemplateWrapper) TemplateIconHandler(apiContext *types.APIContext, next
 		return httperror.NewAPIError(httperror.NotFound, "not found")
 	}
 }
+
+// templateVersionForRancherVersion indicates if a templateVersion works with the rancher server version
+// In the error case it will always return true - if a template is actually invalid for that rancher version
+// API validation will handle the rejection
+func (t TemplateWrapper) templateVersionForRancherVersion(apiContext *types.APIContext, externalID string) bool {
+	var rancherVersion string
+	for query, fields := range apiContext.Query {
+		if query == "rancherVersion" {
+			rancherVersion = fields[0]
+		}
+	}
+
+	if rancherVersion == "" || rancherVersion == "dev" || rancherVersion == "master" {
```

Comment:
This needs updated for our new <release>-head images. Example rancher version: `v2.2-head`.

and head images should should just behave like master: allow all. 

I think you can probably just key off of "ends in -head"

Reply:
Updated